oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/listener/vservice/VirtualCacheProcessor.java

```diff
@@ -0,0 +1,158 @@
+import org.apache.skywalking.oap.server.core.source.Source;
+import org.apache.skywalking.oap.server.core.source.VirtualCacheRead;
+import org.apache.skywalking.oap.server.core.source.VirtualCacheSlowRead;
+import org.apache.skywalking.oap.server.core.source.VirtualCacheSlowWrite;
+import org.apache.skywalking.oap.server.core.source.VirtualCacheWrite;
+import org.apache.skywalking.oap.server.library.util.StringUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+@Slf4j
+@RequiredArgsConstructor
+public class VirtualCacheProcessor implements VirtualServiceProcessor {
+
+    private final NamingControl namingControl;
+
+    private final AnalyzerModuleConfig config;
+
+    private final List<Source> sourceList = new ArrayList<>();
+
+    @Override
+    public void prepareVSIfNecessary(SpanObject span, SegmentObject segmentObject) {
+        if (span.getSpanLayer() != SpanLayer.Cache) {
+            return;
+        }
+        Map<String, String> tags = span.getTagsList().stream()
+                .collect(Collectors.toMap(KeyStringValuePair::getKey, KeyStringValuePair::getValue));
+        if (!(tags.containsKey(SpanTags.CACHE_KEY) && tags.containsKey(SpanTags.CACHE_OP) && tags.containsKey(SpanTags.CACHE_CMD) && tags.containsKey(SpanTags.CACHE_TYPE))) {
+            return;
+        }
+        String cacheType = tags.get(SpanTags.CACHE_TYPE).toLowerCase();
+        String peer = span.getPeer();
+        // peer is blank if it's a local span
+        if (StringUtil.isBlank(peer)) {
+            peer = tags.get(SpanTags.CACHE_TYPE) + "-local";
+        }
+        long timeBucket = TimeBucket.getMinuteTimeBucket(span.getStartTime());
+        String serviceName = namingControl.formatServiceName(peer);
+        int latency = (int) (span.getEndTime() - span.getStartTime());
+        if (latency > 0) {
+            log.info("latency data : {}", new Gson().toJson(span));
+        }
+        sourceList.add(parseServiceMeta(serviceName, timeBucket));
+        String op = tags.get(SpanTags.CACHE_OP);
+        if ("write".equals(op)) {
+            sourceList.add(parseWriteMetrics(span, latency, serviceName, timeBucket));
+            if (latency > config.getCacheWriteLatencyThresholdsAndWatcher().getThreshold(cacheType)) {
+                sourceList.add(parseSlowWrite(segmentObject, latency, serviceName, span, tags));
+            }
+        } else if ("read".equals(op)) {
+            sourceList.add(parseReadMetrics(span, latency, serviceName, timeBucket));
+            if (latency > config.getCacheReadLatencyThresholdsAndWatcher().getThreshold(cacheType)) {
+                sourceList.add(parseSlowRead(segmentObject, latency, serviceName, span, tags));
+            }
+        } else {
+            log.warn("Illegal tag value :{} for 'cache.op' , only write/read would be accepted", op);
```

Comment:
Aren't you having `command` as `op`? And we don't put this warning because there is nothing to warn.

Reply:
You should consider a metric call `cache_access_*` to cover all cases(read/write/others). This would show the access payload more accurate.

Reply:
make sense